### ISSUE TITLE
Tuya account can also be a code instead of Tuya Smart/Life email address

### ISSUE THREAD

**Issue by ZeusFFWD** [home-assistant/home-assistant.github.io#25054]
### Feedback

Under Configuration > Account https://www.home-assistant.io/integrations/tuya#account the guide states your username should be filled. Which is correct as i had 2 accounts where 1 account showed up on the Tuya IoT platform under Devices (see source) as a expected email address, but my other account showed as a long code staring gg-xxxxxxxxx. When using this as my username instead, devices were shown in the HA Tuya integration, where it did not using the email adres i use to configure the other account i had with. You can find your device source(account) here: 

Go to: https://iot.tuya.com/cloud/
Select your project
Go to the Devices tab
Check what is listed as source of your device(s)
This is the Account name you use to setup HA Tuya integration

### URL

https://www.home-assistant.io/integrations/tuya#account

### Version

2022.11

**Comment by ZeusFFWD** [comment]
See also https://github.com/home-assistant/core/issues/80278#issuecomment-1328344746

**Comment by frenck** [comment]
I'm not following, the documentation states: 

`Tuya Smart or Smart Life app account, not your Tuya IoT platform account.`

The title of this documentation feedback makes it looks like we say it is an e-mail address. But we don't state such a thing.

I'm not sure what is wrong or what you are trying to suggest.

../Frenck

**Comment by ZeusFFWD** [comment]
Hi Frenck,

Thanks for getting back at me. The suggestion lies in the fact that it now states
`Tuya Smart or Smart Life app account`
Apparently people (also see [this ](https://github.com/home-assistant/core/issues/80278#issuecomment-1328344746)reply) including me think this "has" to be their email as we use this to sign in to our app, aka "our account". 
My suggestion is to add that this can also be the source code found in the IoT platform in your project _(as my e-mail didn't work, but the code found in the IoT platform did)_
TLDr: When using my app email, the Tuya HA intergrations shows no devices. When using my gg-xxxxxx source as username instead, all works flawlessly (and that is not what i use to sign in with my Tuya app, that is a gmail address).

I'll add a screenshot so you can see what i mean, maybe that clears things up.

![IoT](https://user-images.githubusercontent.com/95367673/204252353-e0e58c55-852b-4535-8e32-30fec846d838.png)

**Comment by frenck** [comment]
> ncluding me think this "has" to be their email as we use this to sign in to our app, aka "our account".

Not sure where email comes from? It doesn't say that. It says: Use the logins you use for the Tuya app. If you are using email in your app, sure.. use email. If you use your phone number, sure use that. Or maybe even an account number?

I'm not sure where the email comes from?

> TLDr: When using my app email, the Tuya HA intergrations shows no devices. When using my gg-xxxxxx source as username instead, all works flawlessly (and that is not what i use to sign in with my Tuya app, that is a gmail address).

☝️ That sounds like a mixup on a different level and the provided procedure is not a fix for that. It is merely working around the actual issue. Please contact Tuya support, it seems you have 2 accounts (probably in different geo-locations).


../Frenck

**Comment by ZeusFFWD** [comment]
Let me say it differently 😅. 

I use email as a account name on my tuya app. 
So I use email in the integration setup correct?

That doesn't work and won't show my entities. 
If I use the weird code found under source for the integration setup, it does work. 

So my point is that either email/username or "source" can be valid options. Where the latter isn't mentioned anywhere.

**Comment by frenck** [comment]
> I use email as a account name on my tuya app.
> So I use email in the integration setup correct?

Correct.

> That doesn't work and won't show my entities.

In that case you might have a different issue with your account. Please contact Tuya support.

**Comment by ZeusFFWD** [comment]
> In that case you might have a different issue with your account. Please contact Tuya support.

No need as it works good like this. Just hoped it would be added as an extra way of signing in which for some people like me was a good work around and saves me a lot of trouble contacting support departments.
